File: 프로그래머스/unrated/138476. 귤 고르기/귤 고르기.js
function solution(k, tangerine) {
    var answer = 0;
    var fre = {};
    var list = []
    for(let i =0;i < tangerine.length;i++){
        if (fre[tangerine[i]] == null){
            fre[tangerine[i]] = 1
        }
        else{
            fre[tangerine[i]] = fre[tangerine[i]] + 1
        }
    }
    for (const value of Object.values(fre)) {
        list.push(value);
    }
    list.sort((a,b) => b - a);
    for(let i = 0;i < list.length ;i++) {
        k -= list[i];
        answer++;
        if(k <= 0){
            break;
        }
    }
    return answer;
}

File: 프로그래머스/unrated/138476. 귤 고르기/귤 고르기.cpp
#include <string>
#include <vector>
#include <map>
#include <iostream>
#include <algorithm>
using namespace std;

int solution(int k, vector<int> tangerine) {
    int answer = 0;
    map<int,int> mymap;
    vector<pair<int,int>> sortmap;
    for(int i = 0;i<tangerine.size();i++){
        if(mymap.find(tangerine[i]) != mymap.end()){
            mymap[tangerine[i]] += 1;
        }
        else{
            mymap[tangerine[i]] = 1;
        }
    }
    for(const auto &pair : mymap){
        sortmap.push_back({pair.second,pair.first});
    }
    sort(sortmap.rbegin(),sortmap.rend());
    for(int i = 0;i<sortmap.size();i++){
        k -= sortmap[i].first;
        answer++;
        if(k <= 0){
            break;
        }
    }
    return answer;
}